### PR TITLE
trying to add brew cask reinstall

### PR DESCRIPTION
**Pull request by pvinis**
I have done some of the work, but I have no idea how to actually run thefuck from the cloned repo. I tried `python thefuck/entrypoints/__init__.py` but that didn't work. I would like to run it so I can test it locally, besides having the tests.

### CODE REVIEW

File: tests/rules/test_brew_cask_reinstall.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import pytest
+from thefuck.types import Command
+from thefuck.rules.brew_cask_reinstall import get_new_command, match
+
+
+output = ("Warning: Cask 'thefuck' is already installed.\n\nTo "
+          "re-install thefuck, run\n  `brew cask reinstall thefuck`")
```
Comment (scorphus) [inline tests/rules/test_brew_cask_reinstall.py]: The actual output is:
```
Warning: Cask 'thefuck' is already installed.

To re-install docker, run:
  brew cask reinstall thefuck
```

```suggestion
          "re-install thefuck, run:\n  brew cask reinstall thefuck")
```

File: tests/rules/test_brew_cask_reinstall.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+@pytest.mark.parametrize('script', [
+    'brew cask reinstall thefuck',
+    'brew install foo'])
+def test_not_match(script):
+    assert not match(Command(script, ''))
+
+
+@pytest.mark.parametrize('script, formula, ', [
+    ('brew install foo', 'foo'),
```
Comment (scorphus) [inline tests/rules/test_brew_cask_reinstall.py]: Missing `cask` here. These tests are failing!

```suggestion
    ('brew cask install foo', 'foo'),
```

File: tests/rules/test_brew_cask_reinstall.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    ('brew install bar zap', 'bar zap')])
```
Comment (scorphus) [inline tests/rules/test_brew_cask_reinstall.py]: Ditto.
```suggestion
    ('brew cask install bar zap', 'bar zap')])
```

File: thefuck/rules/brew_cask_reinstall.py [hunk 4]
```diff
@@ -0,0 +1,17 @@
+import re
+from thefuck.utils import for_app
+
+
+warning_regex = re.compile(r'Warning: Cask \'(?:.(?!is ))+\' is already installed.\n\n')
+message_regex = re.compile(r'To re-install (?:(?!, ).)+, run\n  `brew cask reinstall [^`]+`')
```
Comment (scorphus) [inline thefuck/rules/brew_cask_reinstall.py]: Ditto. Should be changed:
```suggestion
message_regex = re.compile(r'To re-install (?:(?!, ).)+, run:\n  brew cask reinstall [^\n]+')
```

**Comment by pvinis** [comment]
Ah thanks, these makes sense.

I thought about using the same, but since I'm not familiar with the way this works, I thought I'll just copy `brew_reinstall` and adjust it for `cask`.

What I meant with testing was I don't know how to run my forked `thefuck`, so that I can have it suggest the new rule. Basically:
```
> brew cask install thunderbird
Warning: Cask 'thunderbird' is already installed.

To re-install thunderbird, run:
  brew cask reinstall thunderbird
> <Something here so I can run my forked and edited thefuck>
brew cask reinstall thunderbird [enter/↑/↓/ctrl+c]
```

**Comment by scorphus** [comment]
Oh, I read my previous comment again and realized it is incomplete. Sorry.

Assuming you followed the instructions available in the *[Developing](https://github.com/nvbn/thefuck/blob/master/CONTRIBUTING.md#developing)* section of [CONTRIBUTING](https://github.com/nvbn/thefuck/blob/master/CONTRIBUTING.md), the `thefuck` command should point to the version currently checked out when the respective virtual environment is active. In the same way, the `fuck` shell function should also use the current `thefuck`.

So, now for the complete command:

`thefuck brew cask install <formula>`

**Comment by pvinis** [comment]
I think I'm missing something or maybe we can add something in that document.

I clicked on the venv link, I ran `python3 -m venv tutorial-env`, then I run `pip install -r requirements.txt`, then `python setup.py develop`. Everything ran successfully, but when I do `which thefuck` I still get `/usr/local/bin/thefuck`.

I also tried with `python3 -m venv .`.

**Comment by scorphus** [comment]
Did you try with the venv activated?
```bash
python3 -m venv tutorial-env  # create the venv
source tutorial-env/bin/activate  # activate the venv
pip install -r requirements.txt  # install requirements in the venv
python setup.py develop  # install thefuck in the venv
```

**Comment by pvinis** [comment]
I just tried it again and it worked. `which thefuck` gives the correct path. I don't know what's different from what I tried, but it worked, so thanks!
I checked if it corrects the command and it does, so I'm satisfied.

**Comment by scorphus** [comment]
Closing thiis one as `cask` commands are not part of Homebrew anymore.

**Pull request closed without merging**